### ISSUE TITLE
Error messages changed after package upgrade

### ISSUE THREAD

**Issue by creichert** [seagreen/hjsonschema#51]
I noticed a minor oddity when upgrading my stackage resolver recently

```
{
  "type": "object",
  "properties": {
    "foo": {"type":1}
  }
}
```

An error message for an invalid schema changed recently and I'm trying to understand the cause:
- using a check for a completely invalid schema like this: https://github.com/seagreen/hjsonschema/blob/master/examples/Simple.hs#L19
- **before**: `Invalid schema JSON: failed to parse field properties: failed to parse field type: expected [a], encountered Number`
- **after**: `Invalid schema JSON: expected [a], encountered Number`

It's not a huge deal, but it would be nice to have something like the **before** error because syntax errors can be nested inside a very large schema.

Here's the relevant packages I noticed were bumped:

```
-aeson 0.11.2.1
-aeson-pretty 0.8.1
+aeson 1.0.2.1
+aeson-pretty 0.8.2
-hjsonschema 1.6.1
+hjsonschema 1.6.2
```

FWIW, I did investigate changes between `hjsonschema-1.6.1` and `hjsonschema-1.6.2` and did
not notice any changes that would have changed this behavior.

**Comment by seagreen** [comment]
Thanks for the report. I'm traveling at the moment, planning to explore it in detail once I get back.

I'm suspicious that `aeson` caused the change which may make this an instance of a recurring problem-- `aeson` prioritizes speed but in cases like this you want to prioritize error messages. I'm not sure what the solution will be though.

**Comment by seagreen** [comment]
Sorry I never got around to this. We need a JSON library (or alternate options for `aeson`) that focuses on error messages instead of speed, but in the absence of such a lib there's not much we can do here.

(On the off chance someone reads this and does have a suggestion let's definitely reopen the issue)
